gobblin-compaction/src/main/java/gobblin/compaction/HdfsIO.java

```diff
@@ -84,11 +86,15 @@ protected static Configuration getConfiguration() throws FileNotFoundException {
   }
 
   private static void addResourceToConf(Configuration conf) {
-    conf.setStrings(getHdfsUriHadoopPropertyName(), HDFS_URI_DEFAULT);
     addHadoopConfigPropertiesToConf(conf);
+    String hdfsUriKey = getHdfsUriHadoopPropertyName();
     if (CompactionRunner.properties.containsKey(HDFS_URI)) {
-      conf.setStrings(getHdfsUriHadoopPropertyName(), CompactionRunner.properties.getProperty(HDFS_URI));
+      conf.set(hdfsUriKey, CompactionRunner.properties.getProperty(HDFS_URI));
     }
+    if (Strings.isNullOrEmpty(conf.get(hdfsUriKey))) {
```

Comment:
Thanks for your suggestions. I assume that the order of setting ```fs.defaultFS``` is that first we're loading it from the hadoop conf XMLs, then if we see that ```hdfs.uri``` has been defined in the compaction job conf, we use that, and at last we fall back to the default.
If I replace lines 91-93, won't we override the value taken from the hadoop conf with the default one each time if ```hdfs.uri``` is not set?